testgrid/cmd/configurator/prow.go

```diff
@@ -173,32 +173,85 @@ func applySingleProwjobAnnotations(c *configpb.Configuration, pc *prowConfig.Con
 	return nil
 }
 
+// sortPeriodics sorts all periodics by name (ascending).
+func sortPeriodics(per []prowConfig.Periodic) {
+	sort.Slice(per, func(a, b int) bool {
+		return per[a].Name < per[b].Name
+	})
+}
+
+// sortPostsubmits sorts all postsubmits by name and returns a sorted list of org/repos (ascending).
+func sortPostsubmits(post map[string][]prowConfig.Postsubmit) []string {
+	postK := make([]string, 0, len(post))
+
+	for k := range post {
+		postK = append(postK, k)
+	}
+
+	sort.Strings(postK)
+
+	for _, orgrepo := range postK {
+		sort.Slice(post[orgrepo], func(a, b int) bool {
+			return post[orgrepo][a].Name < post[orgrepo][b].Name
+		})
+	}
+
+	return postK
+}
+
+// sortPresubmits sorts all presubmits by name and returns a sorted list of org/repos (ascending).
+func sortPresubmits(pre map[string][]prowConfig.Presubmit) []string {
+	preK := make([]string, 0, len(pre))
+
+	for k := range pre {
+		preK = append(preK, k)
+	}
+
+	sort.Strings(preK)
+
+	for _, orgrepo := range preK {
+		sort.Slice(pre[orgrepo], func(a, b int) bool {
+			return pre[orgrepo][a].Name < pre[orgrepo][b].Name
+		})
+	}
+
+	return preK
+}
+
 func applyProwjobAnnotations(c *configpb.Configuration, reconcile *yamlcfg.DefaultConfiguration, prowConfigAgent *prowConfig.Agent) error {
 	pc := prowConfigAgent.Config()
 	if pc == nil {
 		return nil
 	}
 	jobs := prowConfigAgent.Config().JobConfig
-	for _, j := range jobs.AllPeriodics() {
+
+	per := jobs.AllPeriodics()
+	sortPeriodics(per)
+	for _, j := range per {
 		if err := applySingleProwjobAnnotations(c, pc, j.JobBase, prowapi.PeriodicJob, "", reconcile); err != nil {
 			return err
 		}
 	}
 
-	for repo, js := range jobs.PostsubmitsStatic {
-		for _, j := range js {
-			if err := applySingleProwjobAnnotations(c, pc, j.JobBase, prowapi.PostsubmitJob, repo, reconcile); err != nil {
+	post := jobs.PostsubmitsStatic
+	postK := sortPostsubmits(post)
```

Comment:
naming nit: what does `K` stand for in this context?

alternately, `post := sortPostsubmits(jobs.PostsubmitsStatic)`

Reply:
It signifies `keys` (for the map keys). I will rename it to `postKeys` to be more explicit. Also, `post` ->` map[string][]prowConfig.Presubmit` and `postK` -> `[]string` so I cannot collapse the assignment into one; both references are used later in the loop (`preK` for key ordering and `post` for retrieving the value for that key).